engine/class_modules/sc_warrior.cpp

```diff
@@ -5078,16 +5078,31 @@ void warrior_t::default_apl_dps_precombat()
 
   precombat->add_action( "worldvein_resonance" );
 
-  precombat->add_action( "memory_of_lucid_dreams" );
-
-  precombat->add_action( "guardian_of_azeroth" );
-
   if ( specialization() == WARRIOR_FURY )
   {
     precombat->add_action( this, "Recklessness" );
+   
+    precombat->add_action( this, "potion" );
+    
+    precombat->add_action( this, "memory_of_lucid_dreams" );
+    
+    precombat->add_action( this, "guardian_of_azeroth" );
+  }
+  
+  if ( specialization() == WARRIOR_ARMS )
+  {
+    precombat->add_action( this, "memory_of_lucid_dreams", "if=talent.fervor_of_battle.enabled|!talent.fervor_of_battle.enabled&target.time_to_die>150" );
+    
+    precombat->add_action( this, "guardian_of_azeroth", "if=talent.fervor_of_battle.enabled|talent.massacre.enabled&target.time_to_die>210|"
+                                 "talent.rend.enabled&(target.time_to_die>210|target.time_to_die<145)" );
+    
+    precombat->add_action( this, "potion,name=potion_of_unbridled_fury", "if=essence.condensed_lifeforce.major" );
+    
+    precombat->add_action( this, "potion,name=potion_of_focused_resolve", "if=essence.memory_of_lucid_dreams.major" );
+    
+    precombat->add_action( this, "potion", "if=!essence.memory_of_lucid_dreams.major&!essence.condensed_lifeforce.major" );
   }
 
-  precombat->add_action( "potion" );
```

Comment:
This will remove the usage of potion in precombat for all specs, not just Fury and Arms.

Reply:
The newest commit in this pull request should fix the stated issue, as well as fix issues with the precombat for both Fury and Arms that I overlooked the first time.